lib/trino-geospatial-toolkit/src/main/java/io/trino/geospatial/GeometryUtils.java

```diff
@@ -160,4 +170,26 @@ public static boolean isPointOrRectangle(OGCGeometry ogcGeometry, Envelope envel
 
         return true;
     }
+
+    public static org.locationtech.jts.geom.Geometry jtsGeometryFromJson(String json)
+    {
+        try {
+            return new GeoJsonReader().read(json);
+        }
+        catch (ParseException | IllegalArgumentException e) {
+            throw new TrinoException(INVALID_FUNCTION_ARGUMENT, "Invalid GeoJSON: " + e.getMessage(), e);
+        }
+    }
+
+    public static Optional<String> jsonFromJtsGeometry(org.locationtech.jts.geom.Geometry geometry)
+    {
+        // Empty atomic (non-multi) geometries return null since
+        // there's no way to represent an empty atomic value in GeoJSON
+        if (ATOMIC_GEOMETRY_TYPES.contains(geometry.getGeometryType()) && geometry.isEmpty()) {
+            return Optional.empty();
+        }
+        else { // Empty multi geometries are represented as empty coordinate arrays in GeoJSON
+            return Optional.of(new GeoJsonWriter().write(geometry));
```

Comment:
Jts's `GeoJsonWriter#write` method, for a non-multi typed empty objects returns a malformed JSON `{"type":"Point","coordinates":,"crs":{"type":"name","properties":{"name":"EPSG:0"}}}` which seems like a bug: https://github.com/locationtech/jts/issues/411 So we may need to take it into account if we don't want to serialize as `null`.

Reply:
Had an offline conversation with @martint who suggested just throwing in the case of an empty atomic geometry since the handling is error prone and inconsistent in the library on which we're relying. I'll follow up with them to see if they're open to cleaning this up in which case we can remove this restriction once the library fixes it.